### PR TITLE
refactor: replace secondary colors with warning colors

### LINKED ISSUE #9921: [MOB] Replace secondary colors with warning colors in mobile

### PR DESCRIPTION
**Pull request by brianacnguyen**
## **Description**
This PR replaces the `secondary` colors with `warning` colors, in preparation to upgrade the design-tokens package to v4

## **Related issues**

Fixes: #9921 

## **Manual testing steps**
1. Render the given components
2. Observe changes
3.

### **Before and after**
- `PendingText`
Simulator Screenshot - iPhone 15 Pro - 2024-06-10 at 09 35 40
Simulator Screenshot - iPhone 15 Pro - 2024-06-10 at 09 36 07
- `StyledButtons`
Simulator Screenshot - iPhone 15 Pro - 2024-06-10 at 09 40 08
Simulator Screenshot - iPhone 15 Pro - 2024-06-10 at 09 40 31
- `GasInfoIcon` + `Text`
Simulator Screenshot - iPhone 15 Pro - 2024-06-10 at 09 47 59
Simulator Screenshot - iPhone 15 Pro - 2024-06-10 at 09 48 17

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/Base/StatusText.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ export const PendingText = (props) => {
   return (
     <Text
       bold
-      style={[styles.status, { color: colors.secondary.default }]}
+      style={[styles.status, { color: colors.warning.default }]}
```
Comment (brianacnguyen) [inline app/components/Base/StatusText.js:32]: Simulator Screenshot - iPhone 15 Pro - 2024-06-10 at 09 35 40
Simulator Screenshot - iPhone 15 Pro - 2024-06-10 at 09 36 07

File: app/components/UI/StyledButton/styledButtonStyles.js [hunk 2]
```diff
@@ -20,11 +20,11 @@ const createStyles = (colors) =>
       color: colors.primary.inverse,
     },
     orange: {
-      borderColor: colors.secondary.default,
+      borderColor: colors.warning.default,
```
Comment (brianacnguyen) [inline app/components/UI/StyledButton/styledButtonStyles.js:23]: Simulator Screenshot - iPhone 15 Pro - 2024-06-10 at 09 40 08
Simulator Screenshot - iPhone 15 Pro - 2024-06-10 at 09 40 31

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! Can you please fill out the Manual testing steps for historical context 🙏

File: app/components/Base/StatusText.js (hunk 1 above)
Comment (brianacnguyen) [reply app/components/Base/StatusText.js:32]: this color also fits with the current `PendingText` color in design 
image

**Comment by brianacnguyen** [comment]
> LGTM! Can you please fill out the Manual testing steps for historical context 🙏

not sure how to fill out the manual testing steps since these are deprecated components

**Review by Cal-L — COMMENTED** [review]
LGTM

**Review by wachunei — APPROVED** [review]
LGTM

**Review by georgewrmarshall — APPROVED** [review]

**Review by sleepytanya — APPROVED** [review]

**Pull request merged**